json-schema/json-schema-macros/src/main/scala/endpoints/macros/Macros.scala

```diff
@@ -0,0 +1,201 @@
+package endpoints.macros
+
+import scala.reflect.macros.blackbox
+
+class Macros(val c: blackbox.Context) extends Utils {
+
+  import c.universe._
+
+  final def genericJsonSchemaImpl[A: c.WeakTypeTag]: c.Tree = {
+    val A = weakTypeOf[A]
+    findJsonSchema(A).getOrElse {
+      c.abort(c.enclosingPosition, s"Can't derive JsonSchema[$A]")
+    }
+  }
+
+  private def findJsonSchema(tpe: Type, allowImplicitSearch: Boolean = false): Option[c.Tree] = {
+    {
+      if(primitiveTypes.exists(tpe <:< _) || allowImplicitSearch)
+        findImplicit(tq"JsonSchema[$tpe]")
+      else
+        None
+    }.orElse {
+        if(tpe <:< typeOf[Seq[_]]) {
+          findSeq(tpe)
```

Comment:
Tried this, but stuck at the weird error that I can't understand. I would expect that during implicit  expansion of `arrayJsonSchema` for `JsonSchema[List[GenericSchemas.this.Foo]]`, as required implicit parameter of type `JsonSchema[GenericSchemas.this.Foo]` compiler picks `GenericSchemas.this.Foo.schema`, but for some reason it doesn't.

```
findJsonSchema(GenericSchemas.this.QuuxA)
findJsonSchema(List[GenericSchemas.this.Foo])
findImplicit(JsonSchema[List[GenericSchemas.this.Foo]])
[info] /Users/krzemin/Projects/OpenSource/endpoints/json-schema/json-schema-macros/src/test/scala/endpoints/macros/JsonSchemasTest.scala:69: arrayJsonSchema is not a valid implicit value for GenericSchemas.this.JsonSchema[GenericSchemas.this.Foo] because:
[info] hasMatchingSymbol reported error: polymorphic expression cannot be instantiated to expected type;
[info]  found   : [C[X] <: Seq[X], A]GenericSchemas.this.JsonSchema[C[A]]
[info]  required: GenericSchemas.this.JsonSchema[GenericSchemas.this.Foo]
[info]     implicit val schema: JsonSchema[QuuxA] = genericJsonSchema[QuuxA]
[info]                                                               ^
[info] /Users/krzemin/Projects/OpenSource/endpoints/json-schema/json-schema-macros/src/test/scala/endpoints/macros/JsonSchemasTest.scala:69: arrayJsonSchema is not a valid implicit value for GenericSchemas.this.JsonSchema[List[GenericSchemas.this.Foo]] because:
[info] hasMatchingSymbol reported error: could not find implicit value for parameter jsonSchema: GenericSchemas.this.JsonSchema[GenericSchemas.this.Foo]
[info]     implicit val schema: JsonSchema[QuuxA] = genericJsonSchema[QuuxA]
[info]                                                               ^
implicit found for JsonSchema[List[GenericSchemas.this.Foo]]: <empty>
[info] /Users/krzemin/Projects/OpenSource/endpoints/json-schema/json-schema-macros/src/test/scala/endpoints/macros/JsonSchemasTest.scala:67: Can't derive JsonSchema[List[GenericSchemas.this.Foo]] required for field ss!
[info]   case class QuuxA(ss: List[Foo]) extends Quux
[info]                    ^
[error] /Users/krzemin/Projects/OpenSource/endpoints/json-schema/json-schema-macros/src/test/scala/endpoints/macros/JsonSchemasTest.scala:69: Can't derive JsonSchema[GenericSchemas.this.QuuxA]
[error]     implicit val schema: JsonSchema[QuuxA] = genericJsonSchema[QuuxA]
[error]                                                               ^
```

Pushed the changes with few debug logs (with `-Xlog-implicits` enabled).

Reply:
Finally had some time to investigate it a bit more.

- trying to derive the same instance from REPL made it work well (no code changed!)
- pulling `Foo` implicit json schema instance out of companion object helped as well

To me it seems like some weird compiler edge case, but hopefully it can be easily worked around.

I reorganized structure of example types and their json schemas in tests and it seems that instances for collections are now correctly picked by implicit resolution.